add docs for default org naming (#2984)
clerkinc/clerk-docs: docs/guides/organizations/configure.mdx

Learn more about [Verified Domains](/docs/guides/organizations/add-members/verified-domains).

### Create first organization automatically

You can configure whether Clerk automatically creates the user's first organization. This setting is useful if your application requires every user to belong to an Organization.

Clerk uses the default naming rules to name the automatically created Organization. [Default naming rules](#default-naming-rules) must be enabled to enable this feature.

> 

Reviewer: Agree. Something like this (taken from the same doc page).

<img width="2338" height="636" alt="CleanShot 2026-01-20 at 10 55 28@2x" src="https://github.com/user-attachments/assets/2e9c2da1-94a8-4baa-a06c-14d5a2b0f61e" />

1. Toggle **Create first organization automatically**.

### Default naming rules

Default naming rules automatically suggest an Organization name (and logo, when possible) when users create their first Organization. This feature is required to enable [Create first organization automatically](#create-first-organization-automatically).

If "Create first organization automatically" is disabled, Clerk will prefill the user's first organization name with the result of these naming rules.

Rules are applied in order:

1. **Detect from email domain** automatically populates the Organization name, slug, and logo from the member's email domain (e.g., `alex@clerk.com` → "Clerk").
1. **Personalize from member name** uses member details to create a personalized organization name. You can customize the template using variables like `user.first_name`, `user.last_name`, `user.full_name`, or `user.username` (for example, `{{user.first_name}}'s organization` → "Alex's organization").
1. **Fallback name** is used when no other rule can generate an organization name. By default, this is set to "My organization".

> You can disable email domain detection or member personalization to skip those rules.

Reviewer: ```suggestion
Default naming rules automatically suggest an Organization name (and logo, when possible) when users create their first Organization.

Rules are applied in order:

1. **Detect from email domain** automatically populates the Organization name, slug, and logo from the member's email domain (e.g., `alex@clerk.com` → "Clerk").
1. **Personalize from member name** uses member details to create a personalized organization name. You can customize the template using variables like `user.first_name`, `user.last_name`, `user.full_name`, or `user.username` (for example, `{{user.first_name}}'s organization` → "Alex's organization").
1. **Fallback name** is used when no other rule can generate an organization name. By default, this is set to "My organization".

You can disable email domain detection or member personalization to skip those rules.

> ![NOTE] This feature is required to enable [create first organization automatically](#create-first-organization-automatically). If create first organization automatically is disabled, Clerk will prefill the user's first organization name with the result of these naming rules.
```

You can also [use Organization slugs in your application's URLs](/docs/guides/organizations/org-slugs-in-urls).

### Create first organization automatically

Clerk can automatically create the user's first organization. This is useful if your application requires every user to belong to an Organization.

This setting is disabled by default. To enable it:

1. In the Clerk Dashboard, navigate to the [**Organizations Settings**](https://dashboard.clerk.com/~/organizations-settings) page.
1. Toggle **Create first organization automatically**.

> ![NOTE]

> > Clerk uses [default naming rules](#default-naming-rules) to automatically name the Organization. Default naming rules must also be enabled to use this feature.

Reviewer 2: ```suggestion
> [!NOTE]
> Clerk uses [default naming rules](#default-naming-rules) to automatically name the Organization. Default naming rules must also be enabled to use this feature.
```

### Default naming rules

Default naming rules automatically suggest an Organization name (and logo, when possible) when users create their first Organization.

Rules are applied in order:

1. **Detect from email domain** automatically populates the Organization name, slug, and logo from the member's email domain (e.g., `alex@clerk.com` → "Clerk").
1. **Personalize from member name** uses member details to create a personalized organization name. You can customize the template using variables like `user.first_name`, `user.last_name`, `user.full_name`, or `user.username` (for example, `{{user.first_name}}'s organization` → "Alex's organization").

> 1. **Fallback name** is used when no other rule can generate an organization name. By default, this is set to "My organization".

Reviewer 2: ```suggestion
1. **Fallback name** is used when no other rule can generate an Organization name. By default, this is set to "My organization".
```

You can disable email domain detection or member personalization to skip those rules.

> > [!NOTE] This feature is required to enable [create first organization automatically](#create-first-organization-automatically). If create first organization automatically is disabled, Clerk will prefill the user's first organization name with the result of these naming rules.

Author: ```suggestion
```

These rules can be used on their own or in combination with the [Create first organization automatically](#create-first-organization-automatically) setting:

- When **Create first organization automatically** is enabled, these rules are used to name the automatically created Organization.
- When **Create first organization automatically** is disabled, these rules are used to prefill the Organization name when the user creates their first Organization manually.

Rules are applied in this order:

1. **Detect from email domain** automatically populates the Organization name, slug, and logo from the member's email domain (e.g., `alex@clerk.com` → "Clerk").
1. **Personalize from member name** uses member details to create a personalized Organization name. You can customize the template using variables like `user.first_name`, `user.last_name`, `user.full_name`, or `user.username` (e.g., `{{user.first_name}}'s organization` → "Alex's organization").
1. **Fallback name** is used when no other rule can generate an Organization name. By default, this is set to "My organization".

> You can disable email domain detection or member personalization to skip those rules.

Reviewer 2: ```suggestion
This setting is disabled by default. To enable it:

1. In the Clerk Dashboard, navigate to the [**Organizations Settings**](https://dashboard.clerk.com/~/organizations-settings) page.
1. Toggle **Default naming rules**.
1. Under **Default naming rules**, you can disable email domain detection or member personalization rules individually, if needed.  
```

Maybe have this instead? @manovotny @im2nguyen

Reviewer 2: It's more than for each setting mentioned on this page, we tell the user how to access it and enable it. But we don't have that for this setting currently.

> You can disable email domain detection or member personalization to skip those rules.

Author: i think it's pretty intuitive on how to do this?

You can also [use Organization slugs in your application's URLs](/docs/guides/organizations/org-slugs-in-urls).

### Create first organization automatically

Clerk can automatically create the user's first Organization. This is useful if your application requires every user to belong to an Organization.

This setting is disabled by default. To enable it:

1. In the Clerk Dashboard, navigate to the [**Organizations Settings**](https://dashboard.clerk.com/~/organizations-settings) page.
1. Toggle **Create first organization automatically**.

> [!IMPORTANT]

> > This feature requires the [Default naming rules](#default-naming-rules) setting to be enabled. The naming rules are used to generate the name for the automatically created Organization.

Reviewer 3: this is dashboard related and will need to be handled there - but if it requires "Default naming rules must be enabled." to be enabled, then the setting should not be allowed to be enabled without it. yet, currently, you can enable it and not have "Default naming rules must be enabled." enabled.

Reviewer 2: It's been merged. So will create a PR to remove callout @alexisintech.
